spring-jdbc/src/main/java/org/springframework/jdbc/support/incrementer/AbstractDataFieldMaxValueIncrementer.java

```diff
@@ -133,12 +133,7 @@ public String nextStringValue() throws DataAccessException {
 		String s = Long.toString(getNextKey());
 		int len = s.length();
 		if (len < this.paddingLength) {
-			StringBuilder sb = new StringBuilder(this.paddingLength);
-			for (int i = 0; i < this.paddingLength - len; i++) {
-				sb.append('0');
-			}
-			sb.append(s);
-			s = sb.toString();
+			s = "0".repeat(this.paddingLength - len) + s;
```

Comment:
This may be less efficient due to the need for allocation of a temporary `String`.

Reply:
JMH says otherwise 
<img width="815" alt="изображение" src="https://user-images.githubusercontent.com/38766980/146600554-50fadc17-7c52-4ab5-882d-d1578ba42907.png">

Reply:
Consider attaching benchmark's code